src/Request/Request.php

```diff
@@ -28,4 +31,19 @@ public function getParams(): array
     {
         return $this->params;
     }
+
+    public function getRetryCounter(): int
+    {
+        return $this->retryCounter;
+    }
+
+    public function incrementRetryCounter(): void
+    {
+        $this->retryCounter++;
+    }
+
+    public function isRetryLimitReached(): bool
+    {
+        return $this->retryCounter == self::RETRY_LIMIT;
```

Comment:
We need to make sure to also check the higher counts, in case the increment function is even slightly misused in future changes.

```suggestion
        return $this->retryCounter >= self::RETRY_LIMIT;
```